### PR TITLE
Rosetta-Ravencoin Implementation (PR #2)

### PR DESCRIPTION
**Pull request by ben-abraham**
This PR is a complete functional update to convert the base `coinbase/rosetta-bitcoin` library to work with the ravencoin network.

So far the following segments have been tested/completed. (Note: `check:data` and `check:construction` performed with `rosetta-cli` utility)
- [x] Testnet building
- [x] Testnet indexing
- [x] Testnet `check:data` passing
- [x] Testnet `check:construction` passing
- [x] Mainnet building
- [x] Mainnet indexing
- [x] Mainnet `check:data` passing
- [x] Mainnet `check:construction` passing
- [x] All provided test suites passing
- [x] zstd compression models updated

Shoutout to @HyperPeek and @unclear0122

### CODE REVIEW

File: ravencoin/types.go [hunk 1]
```diff
@@ -79,11 +79,12 @@ const (
...
 // Source: https://bitcoinops.org/en/tools/calc-size/
 const (
-	MinFeeRate            = float64(0.00001) // nolint:gomnd
-	TransactionOverhead   = 12               // 4 version, 2 segwit flag, 1 vin, 1 vout, 4 lock time
-	InputSize             = 68               // 4 prev index, 32 prev hash, 4 sequence, 1 script size, ~27 script witness
-	OutputOverhead        = 9                // 8 value, 1 script size
-	P2PKHScriptPubkeySize = 25               // P2PKH size
+	MinFeeRate                  = float64(0.015) // nolint:gomnd
+	TransactionOverhead         = 12             // 4 version, 2 segwit flag, 1 vin, 1 vout, 4 lock time
+	InputSize                   = 68             // 4 prev index, 32 prev hash, 4 sequence, 1 script size, ~27 script witness
```
Comment (HyperPeek) [inline ravencoin/types.go]: The requirement of MinFeeRate being 0.015 is still not fully understood.

Comment (ben-abraham) [reply ravencoin/types.go]: Fixed! 👍

File: ravencoin/types.go [hunk 2]
```diff
@@ -79,11 +79,12 @@ const (
...
 // Source: https://bitcoinops.org/en/tools/calc-size/
 const (
-	MinFeeRate            = float64(0.00001) // nolint:gomnd
-	TransactionOverhead   = 12               // 4 version, 2 segwit flag, 1 vin, 1 vout, 4 lock time
-	InputSize             = 68               // 4 prev index, 32 prev hash, 4 sequence, 1 script size, ~27 script witness
-	OutputOverhead        = 9                // 8 value, 1 script size
-	P2PKHScriptPubkeySize = 25               // P2PKH size
+	MinFeeRate                  = float64(0.010) // nolint:gomnd
+	TransactionOverhead         = 12             // 4 version, 2 segwit flag, 1 vin, 1 vout, 4 lock time
+	InputSize                   = 148            // 4 prev index, 32 prev hash, 4 sequence, 1 script size, ~107 script witness
```
Comment (HyperPeek) [inline ravencoin/types.go:84]: Thats the fix we needed -- great catch.

**Review by HyperPeek — APPROVED** [review]

**Comment by HyperPeek** [comment]
I did not check the btcd lib on its own (which is more than 70% of the files) as we should trust that. All the rest does not have any obvious issues and is mostly replacing the original implementation with the params for Ravencoin. So approved for merge.

File: ravencoin/types.go (hunk 2 above)
Comment (ben-abraham) [reply ravencoin/types.go:84]: Yup, due to segwit, the input sizes are much larger for us (for now)

**Pull request merged**